hudi-common/src/main/java/org/apache/hudi/common/util/ParquetReaderIterator.java

```diff
@@ -71,24 +66,15 @@ public T next() {
           throw new HoodieException("No more records left to read from parquet file");
         }
       }
-      T retVal = this.next;
-      this.next = read();
```

Comment:
This is wrong, we should not be doing `read()` in next, instead it should be like following:

```
def hasNext() {
  if (next == null) {
    next = read()
  }
  return next != null
}

def next() {
  if (next == null) throw 
  val ret = next
  next = null
  return next
}
```